src/Process/ExecutionLoop.php

```diff
@@ -158,7 +166,40 @@ protected function initialize(): void
         $this->processSet->optimizeOrder($this->optimizeStrategy);
 
         // Initialize first processes that should be run
-        $this->processSet->dequeueProcessesWithoutDelay($this->io);
+        $this->dequeueProcessesWithoutDelay();
+    }
+
+    /**
+     * Set queued processes without delay as prepared
+     */
+    protected function dequeueProcessesWithoutDelay(): void
+    {
+        $queuedProcesses = $this->processSet->get(ProcessWrapper::PROCESS_STATUS_QUEUED);
+
+        foreach ($queuedProcesses as $className => $processWrapper) {
+            if ($processWrapper->isDelayed()) {
```

Comment:
```
if ($processWrapper->isDelayed()) {
...

continue;
}

if ($this->parallelLimitReached()) {
...
continue;
}

$this->io->writeln(
    sprintf('Testcase "%s" is prepared to be run', $className),
    OutputInterface::VERBOSITY_DEBUG
);
$processWrapper->setStatus(ProcessWrapper::PROCESS_STATUS_PREPARED);
```

Reply:
THB I prefer it this way - so the three possible branches are obvious on first sight. Otherwise you must read it whole and be aware of the `continue` statements.